pds_doi_core/util/doi_validator.py

```
@@ -0,0 +1,323 @@
+            if rows[last_row_index][col_index_to_status].lower() not in self.m_workflow_order:
+                logger.error(f"Unexpected status of doi {rows[last_row_index][col_index_to_status].lower()} from database.  Valid values are {self.m_workflow_order.keys()}")
+                raise InvalidDOIException(f"Unexpected status of doi {rows[last_row_index][col_index_to_status].lower()} from database.  Valid values are {self.m_workflow_order.keys()}")
+
+            # Translate the 'status' field into a ranking in the workflow, e.g. 'Draft' to 2
+            rank_current_doi_status = self.m_workflow_order[doi.status.lower()]
+
+            # Translate the 'status' field of same lidvid in database into a ranking in the workflow, e.g. 'Pending' to 3
+            rank_database_doi_status = self.m_workflow_order[rows[last_row_index][col_index_to_status].lower()]
+
+            logger.debug(f"rank_database_doi_status,rank_current_doi_status {rank_database_doi_status,rank_current_doi_status}")
+
+            # A status tuple of ('Pending',3) is higher than ('Draft',2) will cause an error unless --force flag is used.
+            if rank_database_doi_status > rank_current_doi_status:
+                logger.debug(f"rows[last_row_index][col_index_to_status].lower(),doi.status.lower() {rows[last_row_index][col_index_to_status].lower(),doi.status.lower()}")
+                logger.debug(f"CHECK_FIELD_WORKFLOW_GOOD_FALSE: lidvid: {lidvid} status: {doi.status.lower()}")
+                logger.error(f"There is a DOI: {rows[last_row_index][col_index_to_doi]} record which status: '{rows[last_row_index][col_index_to_status].lower()}'.  Are you sure you want to restart the workflow from step: {rank_current_doi_status} for the lidvid: {doi.related_identifier}?")
+                # If the flag --force is set do not throw an exception, otherwise throw an exception to leave the code.
+
+                if force_flag:
```

At this level we should raise the exception whatever the force flag value.

---

The doi_validator.py no longer has force_flag as a parameter.